jujugui/static/gui/src/app/jujulib/charmstore.js

```diff
@@ -447,18 +439,12 @@ var module = module;
           parameter and the response data as its second.
     */
     whoami: function(callback) {
-      var path = this._generatePath('whoami');
-      return jujulib._makeRequest(
-        this.bakery,
-        path,
-        'GET',
-        null,
-        jujulib._transformAuthObject.bind(this, callback),
-        true,
-        // turn off redirect, we want to silently fail
-        // if the macaroon is valid.
-        false
-      );
+      const url = this._generatePath('whoami');
+      const headers = null;
+      const wrappedCallback =
+        jujulib._wrap(
+          jujulib._transformAuthObject.bind(this,callback), {parseJSON: true});
+      return this.bakery.get(url, headers, wrappedCallback);
```

Comment:
This call previously used to "turn off redirect".So I guess we should now use `this.bakery.withoutDischarge().get(...` here.